pkg/inject/sidecar.go

```diff
@@ -134,6 +135,15 @@ func container(jaeger *v1.Jaeger, dep *appsv1.Deployment) corev1.Container {
 		}
 	}
 
+	// Enable tls by default for openshift platform
+	if viper.GetString("platform") == v1.FlagPlatformOpenShift {
+		if len(util.FindItem("--reporter.type=grpc", args)) > 0 && len(util.FindItem("--reporter.grpc.tls=true", args)) == 0 {
+			args = append(args, "--reporter.grpc.tls=true")
+			args = append(args, "--reporter.grpc.tls.ca=/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt")
+			args = append(args, fmt.Sprintf("--reporter.grpc.tls.server-name=%s", service.GetNameForHeadlessCollectorService(jaeger)))
```

Comment:
Same comment as for the agent: needs to qualify the name with namespace + `svc.cluster.local`.